### ISSUE TITLE
Bug: Unexpected error: "`pyarrow.lib.ArrowNotImplementedError`: Cannot write struct type '<field>' with no child field to Parquet."

### ISSUE THREAD

**Issue by bindipankhudi** [airbytehq/PyAirbyte#50]
Following is most likely a side effect of introducing empty tables, related to https://github.com/airbytehq/PyAirbyte/issues/30
```
PyAirbyte % poetry run python examples/run_github.py        

Connection check succeeded for `source-github`.
Started `source-github` read operation at 10:29:46...
Traceback (most recent call last):
  File "/Users/bindipankhudi/PyAirbyte/examples/run_github.py", line 29, in <module>
    result = source.read(cache=ab.new_local_cache("github"))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bindipankhudi/PyAirbyte/airbyte/source.py", line 567, in read
    cache.process_airbyte_messages(
  File "/Users/bindipankhudi/PyAirbyte/airbyte/_processors.py", line 219, in process_airbyte_messages
    self._process_batch(stream_name, record_batch)
  File "/Users/bindipankhudi/PyAirbyte/airbyte/_processors.py", line 238, in _process_batch
    batch_handle = self._write_batch(
                   ^^^^^^^^^^^^^^^^^^
  File "/Users/bindipankhudi/PyAirbyte/airbyte/caches/base.py", line 519, in _write_batch
    return self.file_writer.write_batch(stream_name, batch_id, record_batch)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bindipankhudi/PyAirbyte/airbyte/_file_writers/base.py", line 79, in write_batch
    return self._write_batch(stream_name, batch_id, record_batch)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bindipankhudi/PyAirbyte/airbyte/_file_writers/parquet.py", line 86, in _write_batch
    with parquet.ParquetWriter(output_file_path, schema=record_batch.schema) as writer:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bindipankhudi/PyAirbyte/.venv/lib/python3.11/site-packages/pyarrow/parquet/core.py", line 1016, in __init__
    self.writer = _parquet.ParquetWriter(
                  ^^^^^^^^^^^^^^^^^^^^^^^
  File "pyarrow/_parquet.pyx", line 1869, in pyarrow._parquet.ParquetWriter.__cinit__
  File "pyarrow/error.pxi", line 154, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 91, in pyarrow.lib.check_status
pyarrow.lib.ArrowNotImplementedError: Cannot write struct type 'payload' with no child field to Parquet. Consider adding a dummy child field.
```

**Comment by aaronsteers** [comment]
I just ran into a the same error message when debugging with `source-klaviyo`.

```
pyarrow.lib.ArrowNotImplementedError: Cannot write struct type 'origin_address' with no child field to Parquet. Consider adding a dummy child field.
```

Debugging...

**Comment by bindipankhudi** [comment]
I will prioritize this.

**Comment by bindipankhudi** [comment]
oh nm, didnt realize you are looking into this. thank you! :)

**Comment by aaronsteers** [comment]
Resolved by #60
